gnotes/gnotes_applet.1
.\" This manpage has been automatically generated by docbook2man 
.\" from a DocBook document.  This tool can be found at:
.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
.\" Please send any bug reports, improvements, comments, patches, 
.\" etc. to Steve Cheng <steve@ggi-project.org>.
.TH "GNOTES_APPLET" "1" "20 oktober 2001" "" ""
.SH NAME
gnotes_applet \- Gnotes Applet for the GNOME panel.
.SH SYNOPSIS

\fBgnotes_applet\fR [ \fB--activate-goad-server \fI[=GOAD_ID]\fB\fR] 

.SH "DESCRIPTION"
.PP
\fBgnotes_applet\fR allows you to cover your GNOME
desktop with little, yellow, virtual sticky-notes.
.PP
To add this applet to a Panel, right-click on the Panel and
choose Applets->Utility->GNotes!
.SH "OPTIONS"
.TP
\fB--activate-goad-server\fR
(Internal use only) GOAD server ID to activate.

GOAD_ID is usually the applet name [\fBgnotes_applet\fR]
.SH "AUTHOR"
.PP
This manual page was written by Christian Marillat <marillat@debian.org> for
the Debian GNU/Linux system (but may be used by others).


NEWS
News for gnome-applets-1.4.0.2 release:

- general

	* Portability fixes from <drk@sgi.com>. (Kjartan)
	
- asclock

	* Portability fixes from <mcnichol@austin.ibm.com> (Kjartan)
	
- clockmail

	* New theme (gree-digital-analog) from Alex Boxley
	  <alex.bowley@excitehome.net>
	* Portability fixes from <mcnichol@austin.ibm.com> (Kjartan)

- drivemount

	* Fix buffer overruns in sscanf() usage. From John Ellis
	  <johne@bellatlantic.net>. (Kjartan)
	  
- fifteen

	* Portability fixes. (Kjartan)
	
- geyes

	* Spanish translation of the docs. (Manuel)
	* Compile fixes on Irix from <drk@sgi.com>. (Kjartan)
	* Compile fixes in AIX from <mcnichol@austin.ibm.com>. (Kjartan)
	
- gkb

	* Fix Czech flag. (Pavel)
	* Warning/portability fixes. (Kjartan)

- gweather

	* Compile fixes for AIX from <mcnichol@austin.ibm.com>. (Kjartan)
	* Patch from <mdoller@wpi.edu> to make detailed forecast more readable.
	* Round off temperatures correctly. Patch from bugzilla. (Kjartan)
	
- life

	* Compile fixes for AIX from <mcnichol@austin.ibm.com>. (Kjartan)

- mini-commander

	* Fix bug which caused themes not to draw the entry border correctly
	  from Norman Stevens <norman@arcady.u-net.com>. (Kjartan)

- mixer

	* Fix to work with wheel-mice. Patch from Mike Kelly <mike@csuchico.edu>
	* Compile fixes for Irix from David KAELBLING <drk@sgi.com>. (Kjartan)
	
- odometer

	* Fix typos in properties. (Stanislav)

- slashapp

	* Patch from Richard Kinder <r_kinder@yahoo.com> which fixes a whole
	  lot of bugs where things crash on updating and article reading.
	  (Kjartan)
	* Fix memory leaks and robustness issues. (George)
	
- tickastat

	* Fix for compilation on Irix from David KAELBLING <drk@sgi.com>.
	


gnotes/gnotes_applet.c
/*
#
#   GNotes!
#   Copyright (C) 1998-1999 spoon <spoon@ix.netcom.com>
#   Copyright (C) 1999 dres <dres@debian.org>
#
#   This program is free software; you can redistribute it and/or modify
#   it under the terms of the GNU General Public License as published by
#   the Free Software Foundation; either version 2 of the License, or
#   (at your option) any later version.
#
#   This program is distributed in the hope that it will be useful,
#   but WITHOUT ANY WARRANTY; without even the implied warranty of
#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#   GNU General Public License for more details.
#
#   You should have received a copy of the GNU General Public License
#   along with this program; if not, write to the Free Software
#   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
*/

#include <config.h>

#include <sys/types.h>
#include <sys/stat.h>
#include <gnome.h>
#include <applet-widget.h>
#include <signal.h>

#include "gnotes_applet.h"
#include "gnotes.xpm"
#include "gnotes_session.h"
#include "gnote.h"

GNotes gnotes;

GNotes *gnotes_get_main_info(void)
{
    return &gnotes;
}

/* what happens when a button is pressed */
static gint applet_button_press_cb(GtkWidget *widget, GdkEventButton *event,
                                   gpointer data)
{
    switch(event->button)
    {
    case 1:
        gnote_new_cb(APPLET_WIDGET(gnotes.applet), 0);
        break;
    default:
        return(FALSE);
    };
    return(TRUE);
};

static GtkWidget *create_gnotes_button(GNotes *gnotes)
{
    GtkWidget *frame;
    GtkWidget *vbox;
    int size;

    frame = gtk_frame_new(NULL);
    gtk_frame_set_shadow_type(GTK_FRAME(frame), GTK_SHADOW_IN);
    gtk_widget_show(frame);

    vbox = gtk_vbox_new(FALSE, FALSE);
    gtk_container_add(GTK_CONTAINER(frame), vbox);
    gtk_widget_show(vbox);

    gnotes->button = gtk_button_new();
    GTK_WIDGET_UNSET_FLAGS(gnotes->button, GTK_CAN_DEFAULT);
    GTK_WIDGET_UNSET_FLAGS(gnotes->button, GTK_CAN_FOCUS);
    size = applet_widget_get_panel_pixel_size(APPLET_WIDGET(gnotes->applet));
    gnotes->pixmap = gnome_pixmap_new_from_xpm_d_at_size(gnotes_xpm, 
							 (size*40)/48,
							 (size*40)/48);
    gtk_box_pack_start(GTK_BOX(vbox), gnotes->button, FALSE, TRUE, 0);
    gtk_widget_show(gnotes->pixmap);
    gtk_container_add(GTK_CONTAINER(gnotes->button), gnotes->pixmap);
    gtk_widget_show(gnotes->button);

    gtk_signal_connect(GTK_OBJECT(gnotes->button), "button_press_event",
                       GTK_SIGNAL_FUNC(applet_button_press_cb), NULL);

    return(frame);
};

static void about(AppletWidget *applet, gpointer data)
{
    static const char *authors[] = { "spoon <spoon@ix.netcom.com>",
                                     "dres <dres@debian.org", NULL };
    GtkWidget *about_box;

    about_box = gnome_about_new(_("GNotes!"), VERSION,
                                _("Copyright (C) 1998-1999 spoon <spoon@ix.netcom.com> \nCopyright (C) 1999 dres <dres@debian.org>"),
                                authors,
                                _("Create sticky notes on your screen."), NULL);

    gtk_widget_show(about_box);
}


static gint applet_save_session(GtkWidget *widget, char *privcfpath,
                                char *globcfpath, gpointer data)
{
    g_debug("Saving session\n");
    gnotes_preferences_save(privcfpath, &gnotes);
    gnotes_save(0, 0);
    return FALSE;
}

static void applet_change_pixel_size(GtkWidget *widget, int size,
				     gpointer data)
{
    GNotes *gnotes = data;
    g_debug("Changing pixel size to: %d\n", size);
    gtk_widget_destroy(gnotes->pixmap);
    gnotes->pixmap = gnome_pixmap_new_from_xpm_d_at_size(gnotes_xpm, 
							 (size*40)/48,
							 (size*40)/48);
    gtk_widget_show(gnotes->pixmap);
    gtk_container_add(GTK_CONTAINER(gnotes->button), gnotes->pixmap);
}


int main(int argc, char **argv)
{
    GtkWidget *gnotes_button;
    GnomeClient *smClient;

    /* set up the GNotes directory */
    umask(0077);
    mkdir(get_gnotes_dir(), 0700);
    chmod(get_gnotes_dir(), 0700);
    chdir(get_gnotes_dir());
    
    bindtextdomain(PACKAGE, GNOMELOCALEDIR);
    textdomain(PACKAGE);

    /* initialize the gnotes structures */
    gnotes_init();

    applet_widget_init("gnotes_applet", VERSION, argc, argv, NULL, 0, NULL);

    smClient = newGnomeClient();        /* initialize session management */

    gnotes.applet = applet_widget_new("gnotes_applet");

    gnotes.defaults.onbottom = FALSE;
    
    if (!gnotes.applet)
    {
        g_error(_("Can't create GNotes applet!"));
        exit(1);
    }

    gnotes_button = create_gnotes_button(&gnotes);
    applet_widget_add(APPLET_WIDGET(gnotes.applet), gnotes_button);

    gtk_widget_show(gnotes_button);
    gtk_widget_show(gnotes.applet);

    applet_widget_register_stock_callback(APPLET_WIDGET(gnotes.applet),
                                          "about", GNOME_STOCK_MENU_ABOUT,
                                          _("About..."), about, NULL);
    applet_widget_register_stock_callback(APPLET_WIDGET(gnotes.applet),
                                          "properties", GNOME_STOCK_MENU_PROP,
                                          _("Properties..."), properties_show,
                                          &gnotes);

    applet_widget_register_callback(APPLET_WIDGET(gnotes.applet), GNOTES_RAISE,
                                    _(GNOTES_RAISE), gnotes_raise, 0);
    applet_widget_register_callback(APPLET_WIDGET(gnotes.applet), GNOTES_LOWER,
                                    _(GNOTES_LOWER), gnotes_lower, 0);
    applet_widget_register_callback(APPLET_WIDGET(gnotes.applet), GNOTES_HIDE,
                                    _(GNOTES_HIDE), gnotes_hide, 0);
    applet_widget_register_callback(APPLET_WIDGET(gnotes.applet), GNOTES_SHOW,
                                    _(GNOTES_SHOW), gnotes_show, 0);
    /*
      applet_widget_register_stock_callback(APPLET_WIDGET(gnotes.applet),
      GNOTES_SAVE, GNOME_STOCK_PIXMAP_SAVE, _(GNOTES_SAVE), 
      GTK_SIGNAL_FUNC(gnotes_preferences_save_cb), NULL);
    */

    /* let's load us some notes! */
    gnotes_load(0, 0);

    /* load up the preferences */
    gnotes_preferences_load(APPLET_WIDGET(gnotes.applet)->privcfgpath,
                            &gnotes);
    
    gtk_signal_connect(GTK_OBJECT(gnotes.applet), "change_pixel_size",
                       GTK_SIGNAL_FUNC(applet_change_pixel_size), &gnotes);
    gtk_signal_connect(GTK_OBJECT(gnotes.applet), "save_session",
                       GTK_SIGNAL_FUNC(applet_save_session), NULL);

    applet_widget_gtk_main ();

    return(0);
};

